lib/p2p/packets/types/SwapCompletePacket.ts

```diff
@@ -0,0 +1,19 @@
+import Packet, { PacketDirection } from '../Packet';
+import PacketType from '../PacketType';
+
+export type SwapCompletePacketBody = {
+  r_hash: string;
+  r_preimage?: string;
```

Comment:
Why is this field optional? If the swap is complete, shouldn't the preimage be shared knowledge at this point?

Reply:
WIP. Not there anymore.